models/Document/Listing.php

```diff
@@ -131,4 +131,30 @@ public function getItems($offset, $itemCountPerPage)
 
         return $this->load();
     }
+
+
+    /**
+     * @param Model\User $user
+     * @param string $permission
+     *
+     * @return static
+     * @internal
+     */
+    public function filterAccessibleByUser(Model\User $user, $permission = 'list')
+    {
+        if (!$user->isAdmin()) {
+            $userIds = $user->getRoles();
+            $userIds[] = $user->getId();
+
+            $condition = '(
+                (SELECT `'.$permission.'` FROM users_workspaces_document WHERE userId IN ('.implode(',', $userIds).') AND LOCATE(CONCAT(path,`key`),cpath)=1 ORDER BY LENGTH(cpath) DESC, FIELD(userId, '.$user->getId().') DESC, `'.$permission.'` DESC LIMIT 1)=1
```

Comment:
same here, why not use `$this->quote()`

Reply:
Let's discuss in above conversation. When we find another method there, I will apply it also here.